lib/guard/falcon/controller.rb

```diff
@@ -77,7 +77,7 @@ def run_server
 				
 				logger.info("Starting Falcon HTTP server on #{endpoint}.")
 				
-				container = Async::Container::Forked.new(concurrency: @options[:concurrency]) do
+				container = Async::Container::Forked.new do
```

Comment:
You need to change it to something like:

```ruby
container = Async::Container::Forked.new(count: @options[:count])

container.run do ...
    # existing code
end
```

Reply:
It's `container.run` that takes the `count` option. However when I do that I get:

```
Guard::Falcon::Controller
22:56:10 - INFO - Starting Falcon HTTP server on #<Falcon::Endpoint http://localhost:9292/>.
  0.1s: <Async::Task:0x2af81b8d4490 failed>
      |  IOError: closed stream
      |  → /home/huba/.gem/gems/async-1.15.5/lib/async/wrapper.rb:53 in `initialize_copy'
      |    /home/huba/.gem/gems/async-1.15.5/lib/async/wrapper.rb:53 in `initialize_dup'
      |    /home/huba/.gem/gems/async-1.15.5/lib/async/wrapper.rb:53 in `dup'
      |    /home/huba/.gem/gems/async-1.15.5/lib/async/wrapper.rb:53 in `dup'
      |    /home/huba/.gem/gems/async-io-1.20.0/lib/async/io/generic.rb:101 in `dup'
      |    /home/huba/.gem/gems/async-io-1.20.0/lib/async/io/shared_endpoint.rb:70 in `block in bind'
      |    /home/huba/.gem/gems/async-io-1.20.0/lib/async/io/shared_endpoint.rb:69 in `each'
      |    /home/huba/.gem/gems/async-io-1.20.0/lib/async/io/shared_endpoint.rb:69 in `bind'
      |    /home/huba/.gem/gems/async-io-1.20.0/lib/async/io/shared_endpoint.rb:103 in `accept'
      |    /home/huba/.gem/gems/async-http-0.38.1/lib/async/http/server.rb:69 in `run'
      |    ./lib/guard/falcon/controller.rb:95 in `block in run_server'
      |    /home/huba/.gem/gems/async-1.15.5/lib/async/task.rb:199 in `block in make_fiber'
  0.1s: #<Async::Container::Forked:0x000055f037180420>
      | Unnamed pid 10084 exit 0
  should start server

Finished in 0.0099 seconds (files took 0.20172 seconds to load)
1 example, 0 failures
```

Reply:
I see what is happening. You are closing the bound endpoint before calling `container.wait`. Sorry, this also changed a bit. I wonder if it makes it impossible to use.

Reply:
It's probably super confusing but `container.run` doesn't actually run any process. It schedules it to run, and it only runs when you call `container.wait`.

Reply:
So the endpoint appears to be bound in its own reactor before the container is created.

Reply:
Yeah, that doesn't matter the shared endpoint takes care of that.

Reply:
`Container#wait` is blocking to ensure child process doesn't get borked...

Can you push what you've done and I will sort it out.

Reply:
This is a cross-repo issue now lol.

Reply:
One quick fix you could try is to use Async::Container::Threaded... it doesn't have the same semantics yet, it creates threads as soon as you call run.